### PR TITLE
Feature/tri 419

### PR DESCRIPTION
**Pull request by ds-ext-kmassalski**
- added OAuth2 interceptor to RestTemplate - every request contains access token in Authorization header.
- access token is automaticaly refreshed after expiration (5min  lifetime)
- replaced Feign to RestTemplate for DTR and Submodel Clients for reusage of above interceptor, and common place for Oauth
- our API stays public as before, I will secure it in scope of other task (TRI-420)
- added some loggers to not be blind in case of errors

### CODE REVIEW

**Comment by ds-ext-kmassalski** [comment]
CodeQL problem (.csrf().disable()) - it seems to be false-positive for me.
Right now API is public, and in the future it will be stateless, secured by JWT token, without any session.
@ds-jkreutzfeld please take a look

**Comment by ds-jkreutzfeld** [comment]
I don't see the need for CSRF protection in this case as well. We can suppress this warning.

File: irs-api/src/main/resources/application.yml [hunk 1]
```diff
@@ -4,6 +4,17 @@ server:
...
+    oauth2:
+      client:
+        registration:
+          keycloak:
+            authorization-grant-type: client_credentials
+            client-id: ${KEYCLOAK_OAUTH2_CLIENT_ID}
+            client-secret: ${KEYCLOAK_OAUTH2_CLIENT_SECRET}
+        provider:
+          keycloak:
+            token-uri: ${KEYCLOAK_OAUTH2_CLIENT_TOKEN_URI}
```
[the remark this answers is not in the document]
Comment (ds-ext-kmassalski) [reply irs-api/src/main/resources/application.yml:17]: @ds-jkreutzfeld please check now - I deployed on dev with vault from my branch with success, so I think its working.

[the remark this answers is not in the document]
Comment (ds-jkreutzfeld) [reply irs-api/src/main/resources/application.yml:17]: Looks good now :)

**Review by ds-jkreutzfeld — COMMENTED** [review]
Apart from the failing sonar build, everything is fine 👍

**Review by ds-jkreutzfeld — APPROVED** [review]

**Pull request merged**